lib/game.rb

```diff
@@ -13,16 +13,41 @@ def initialize(input:, output:)
   end
 
   def play
+    introduce_the_game
+    play_to_completion
+    bid_farewell
+  end
+
+  private
+
+  attr_reader :grid, :input, :output
+
+  def introduce_the_game
     grid_view.render
+  end
+
+  def play_to_completion
+    until grid.full?
```

Comment:
This loop matches the loop of getting and validating feedback, in that I also struggled getting it setup and I don't feel very good about the end result.

The pattern seems to be: do X and Y until Z happens, but Z will only ever happen between X and Y. 🤔

Reply:
From a readability point of view it looks good. Is your worry that the player will fill the last grid spot, and then the computer is doing an unneeded bit of work?

I'm not so bothered by it if these methods knows how to deal with that scenario. In fact I really like the simplicity of counting on these methods to take care of edge cases on their own so that it can look that simple here.

Reply:
I think I'm just getting bothered that the loop conditional and the inner conditional are the same. It would be nice to have that conditional in one place and the loop cycles between player and computer. I'm not going to fixate on it now, though. Maybe if a future requirement is to support two players, that would push the design in some direction.